client/components/webpack-build-monitor/index.jsx

```diff
@@ -102,17 +102,17 @@ const RELOAD_REQUIRED_ELEMENT = (
 	<div className="webpack-build-monitor is-warning">Need to refresh</div>
 );
 
-class WebpackBuildMonitor extends React.Component {
+class WebpackBuildMonitor extends React.PureComponent {
 	state = {
 		buildStatus: IDLE,
 		reloadRequired: false,
 	};
 
 	componentDidMount() {
 		this.unsubscribe = store.subscribe( () => {
-			const status = store.getState();
-			if ( ! isEqual( status, this.state ) ) {
-				this.setState( store.getState() );
+			const state = store.getState();
+			if ( state !== this.state ) {
```

Comment:
Sorry for the bad news, but this comparison still doesn't do what you would expect :wink:

The `state` returned from `store.getState()` is the Redux state. If the dispatched action didn't change the state, it's guaranteed to be `===`-equal to the previous Redux state.

But `this.state` is a very different object -- it's the React component's internal state and its value was created by merging the argument to `this.setState` into the previous value. It's a different JS object.

If you'd like to simulate the perf optimization that `connect()` does internally, you'd need to compare Redux state to another Redux state:

```js
store.subscribe( ( function() {
  let prevState = undefined;
  return () => {
    const nextState = store.getState();
    if ( nextState !== prevState ) {
      this.setState( nextState );
      prevState = nextState;
    }
  }
} )() );
```

Reply:
that makes total sense.  
The old one would have worked with `isEqual` but been less efficient.

I'll switch to your implementation

Reply:
OK, thanks, let's merge now 👍 

Note that we probably are over-optimizing here -- even if we called `setState` on every call of the store listener, the `PureComponent` would do a shallow state comparison and avoid rerender anyway.